src/coreclr/jit/emit.cpp

```diff
@@ -3811,10 +3821,21 @@ void emitter::emitDispIG(insGroup* ig, insGroup* igPrev, bool verbose)
                 {
                     instrDesc* id = (instrDesc*)ins;
 
+#if defined(TARGET_XARCH)
+                    if (emitInstHasNoCode(id))
+                    {
+                        // an instruction with no code prevents us being able to iterate to the
+                        // next instructions so we must be certain that when we find one it is
```

Comment:
Hhm, this will stop displaying the `align` instruction as well? We want to continue displaying `align` instruction though.

Reply:
Example:

From benchmarks collection of windows/x64: 13400.dasm

![image](https://user-images.githubusercontent.com/12488060/171868257-5f33e9cc-2b37-4d9f-8b98-91fb15296f21.png)

Reply:
I was hoping to see all improvements, but there is 1 odd case of regression (the same method as above) in benchmark collection that shows regression. The culprit is we are not calculating the `igSize` of one of the IG correctly I believe.

![image](https://user-images.githubusercontent.com/12488060/171868784-112a18e9-3875-4da6-9eae-405c954ec11c.png)

Reply:
Unrelated, but there is something odd going on with the "Total code size" being reported in the disasm. If I sum up all the `size=` values in base, their total is 7868 bytes, but we are reporting it as 7827 bytes. I will investigate.

[13400_diff.dasm.md](https://github.com/dotnet/runtime/files/8832890/13400_diff.dasm.md)
[13400_base.dasm.md](https://github.com/dotnet/runtime/files/8832892/13400_base.dasm.md)

Reply:
Seems like the `size=` displayed is the allocated size and not the actual size which might differ because of instruction size over-estimation. If you don't mind, can you fix to report the true size? It should be ~ig->igSize~ (cp - bp)

https://github.com/dotnet/runtime/blob/b44fd4aeffafd68bc710b2feec66318835ca3edc/src/coreclr/jit/emit.cpp#L6620

Reply:
I've changed the condition to `if ((id->idIns() != INS_align) && emitInstHasNoCode(id))` so align is specifically allowed through.

I was suspicious of the sizes but the small number of examples I worked through all added up. a 5 byte max size jmp was removed replacing what would have been bound as a 2 byte jmp leaving a 2 byte reduction overall. I didn't hit any aligned ig's though because i was looking at diffs and those aligned groups wouldn't have been changed.

Size is now calculated from the pointers not reported from the ig as requested.

Reply:
I would change this to `id->idIns() != INS_jmp` instead because you are doing this explicitly for `jmp`.

Reply:
> but the small number of examples I worked through all added up

Lets keep an eye on benchmarks diffs from `superpmi-diffs` pipeline and spot the regression.

Reply:
> I would change this to `id->idIns() != INS_jmp` instead because you are doing this explicitly for `jmp`.

Ok, done.

Reply:
> Ok, done.

I meant `id->idIns() == INS_jmp`

Reply:
Also seems like arm64 builds are failing, so you might want to fix them before doing another push.

Reply:
An odd failure. I wouldn't expect an ARM64 build to be TARGET_XARCH which is what protected both the function and the call site. SingleAccretion suggested removing the inline from the implementation so I've pushed that to see if it helps.